tests/t330-basis.h

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2017-2018, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory. LLNL-CODE-734707.
+// All Rights reserved. See files LICENSE and NOTICE for details.
+//
+// This file is part of CEED, a collection of benchmarks, miniapps, software
+// libraries and APIs for efficient high-order finite element and spectral
+// element discretizations for exascale applications. For more information and
+// source code availability see http://github.com/ceed.
+//
+// The CEED research is supported by the Exascale Computing Project 17-SC-20-SC,
+// a collaborative effort of two U.S. Department of Energy organizations (Office
+// of Science and the National Nuclear Security Administration) responsible for
+// the planning and preparation of a capable exascale ecosystem, including
+// software, applications, hardware, advanced system engineering and early
+// testbed platforms, in support of the nation's exascale computing imperative.
+
+// Hdiv basis for quadrilateral linear BDMelement in 2D
+// Local numbering is as follow (each edge has 2 vector dof)
+//     b4     b5
+//    2---------3
+//  b7|         |b3
+//    |         |
+//  b6|         |b2
+//    0---------1
+//     b0     b1
+// Bx[0-->7] = b0_x-->b7_x, By[0-->7] = b0_y-->b7_y
+// To see how the nodal basis is constructed visit:
+// https://github.com/rezgarshakeri/H-div-Tests
+int NodalHdivBasisQuad(CeedScalar *X, CeedScalar *Bx, CeedScalar *By) {
+  CeedScalar xhat = X[0];
```

Comment:
we use `x_hat` over `xhat` typically

Reply:
Yeah. And since H^1 is contained in H(div), should we automatically construct it from the gradient if requested (and number of components = dimension)? You tend to use H^1 and H(div) elements differently (Piola map) so this might cause more harm than good (say, in evaluating the off-diagonal block in mixed velocity-pressure).

Reply:
> we use `x_hat` over `xhat` typically

Thanks.

Reply:
> 

So you mean add `div` to H^1 constructor for the case when we need the divergence in H^1?! Isn't the sum of the diagonal of gradient?!

Reply:
I think I'm making the case that even though divergence appears in weak forms when using H^1 elements (as in `q * div(u) + p * div(v)` for Stokes or mixed elasticity), we should not create it by default because it can't be used on stretched elements (reference element divergence doesn't tell you physical element divergence for these elements). So let's refuse to emulate in this way that would likely only cause confusing bugs.

Reply:
Should we leave `CEED_EVAL_DIV` for H^1 in another PR? perhaps when we do incompressible elasticity?!

Reply:
I think `CEED_EVAL_DIV` should probably error for H^1, at leas for now. So yes, not this MR.

Reply:
Though, it seems reasonable to me to collapse down to a single backend BasisApply by introducing this c/k division below. I think that would give us a smaller about of code, but a wider range of cases we could handle. In the H1 case we can error because no div matrix is set without knowing specifically that we're doing a H1 Apply

Reply:
Yeah, right now we have `CEED_EVAL_DIV not supported` for H^1. I can add switch case that div works for H(div) only.